### ISSUE TITLE
Globus tokens confusion

### ISSUE THREAD

**Issue by blawney** [web-mev/mev-backend#133]
Issue:
If a WebMeV user has previously authenticated with Globus using their Globus account identified by X, we have globus tokens in our DB. If that same WebMeV account later attempts a Globus transfer while logged into globus using ANOTHER identity (Y), then our attempt to refresh the X-associated tokens fails. The stack trace appears as follows:
```
INFO 2023-03-28 19:06:47,277 base: Validating token
INFO 2023-03-28 19:06:47,494 requests: request done (success)
INFO 2023-03-28 19:06:47,494 globus: Token was not active. Go refresh.
INFO 2023-03-28 19:06:47,495 base: Executing token refresh; typically requires client credentials
INFO 2023-03-28 19:06:47,495 base: Fetching new token from Globus Auth
INFO 2023-03-28 19:06:47,555 requests: request done (success)
INFO 2023-03-28 19:06:47,556 globus: Encountered error with Globus token refresh. Code was 400
Message was: {"error":"invalid_grant"}
INFO 2023-03-28 19:06:47,839 globus: Failed to update or get active tokens for Globus.
INFO 2023-03-28 19:06:47,839 globus_views: Did not have a recent authentication/session. Send to Globus auth.
INFO 2023-03-28 19:06:47,839 globus: Get Globus token for user brianlawney@gmail.com
INFO 2023-03-28 19:06:47,841 basic: Setting up a BasicAuthorizer. It will use an auth type of Basic and cannot handle 401s.
INFO 2023-03-28 19:06:47,841 basic: BasicAuthorizer.username = be033cd4-c6d0-4b44-8bed-30b5d6448107
INFO 2023-03-28 19:06:47,841 client: Creating client of type <class 'globus_sdk.services.auth.client.confidential_client.ConfidentialAppAuthClient'> for service "auth"
INFO 2023-03-28 19:06:47,842 confidential_client: Finished initializing client, client_id=be033cd4-c6d0-4b44-8bed-30b5d6448107
INFO 2023-03-28 19:06:47,842 confidential_client: Checking token validity (introspect)
INFO 2023-03-28 19:06:48,040 requests: request done (success)
INFO 2023-03-28 19:06:48,040 globus: Token data from introspect:
{
  "active": false
}
ERROR 2023-03-28 19:06:48,043 log: Internal Server Error: /api/globus/initiate/
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.10/dist-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/django/views/generic/base.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.10/dist-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.10/dist-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.10/dist-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/srv/mev-backend/mev/api/views/globus_views.py", line 282, in get
    return self.handle_session_condition(client,
  File "/srv/mev-backend/mev/api/views/globus_views.py", line 215, in handle_session_condition
    globus_user_uuid = get_globus_uuid(user)
  File "/srv/mev-backend/mev/api/utilities/globus.py", line 131, in get_globus_uuid
    return introspection_data['sub']
KeyError: 'sub'
```

**Comment by blawney** [comment]
Related when debugging this on dev server:

Until the transfer is submitted, WebMeV has no idea which Globus identity is currently authenticated/logged in in your browser. The backend takes the existing cached Globus tokens, ensures they are current, possibly updates, etc. and then submits the transfer with the transfer info (i.e. the files to transfer) which comes from the frontend POST request.

However, if the Globus tokens we have are for globus identity X and the current browser session has globus identity Y, the transfer submission fails with the following error:

```
AuthorizationParameterInfo(session_message=Session reauthentication required (Globus Transfer), session_required_identities=[], session_required_single_domain=['harvard.edu'], session_required_policies=None)
```

**Comment by blawney** [comment]
The first error reported is likely due to changing the endpoint on the server (the tokens which are in the database are related to a globus endpoint that no longer exists). This error is moot.

The latter error has been addressed in 75426adaab87072057f912b396fa389bf120ddfb
